src/fetch-map/layer-map.ts

```diff
@@ -39,7 +39,24 @@ import type {ProviderType} from '../types.js';
 import {DEFAULT_AGGREGATION_EXP_ALIAS} from '../constants-internal.js';
 import type {SchemaField} from '../types-internal.js';
 
-const SCALE_FUNCS: Record<string, () => any> = {
+export type D3Scale = {
+  domain: (d?: any) => any[];
+  range: (d?: any) => any[];
+  unknown?: (d?: string) => any;
+} & ((d: any) => any);
+type D3ScaleFactory = () => D3Scale;
+
+export type SCALE_TYPE =
```

Comment:
For consistency let's use `ScaleType` (title case) to refer to a type union or enum.